src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountUpdate.java

```diff
@@ -2,20 +2,94 @@
 
 import com.yahoo.identity.IdentityException;
 import com.yahoo.identity.services.account.AccountUpdate;
+import org.apache.ibatis.session.SqlSession;
 import org.apache.ibatis.session.SqlSessionFactory;
 
 import javax.annotation.Nonnull;
+import java.time.Instant;
 
 public class SqlAccountUpdate implements AccountUpdate {
     private final SqlSessionFactory sqlSessionFactory;
+    private AccountModel account;
 
-    public SqlAccountUpdate(@Nonnull SqlSessionFactory sqlSessionFactory) {
+    public SqlAccountUpdate(@Nonnull SqlSessionFactory sqlSessionFactory, @Nonnull String id) {
         this.sqlSessionFactory = sqlSessionFactory;
+
+        try(SqlSession session = sqlSessionFactory.openSession()){
+            AccountMapper mapper = session.getMapper(AccountMapper.class);
+            this.account = mapper.getAccount(id);
+        }catch (Exception e) {
+            this.account = new AccountModel();
+        }
```

Comment:
Why we need to get the account from db first before updating?

Reply:
Sorry I didn't discuss the implementation details with you. 

Idea: get the old email, password, description -> replace some of them with new ones -> write in a unified SQL operation: `UPDATE account SET email=email, password=password, description=description.`

Do you mean that we can assume the request sent by frontend must contain all three editable fields? If yes, we can drop the getAccount() method.

Reply:
Not all fields would be available every time.
To ignore null properties using MyBatis, see http://www.mybatis.org/mybatis-3/dynamic-sql.html

Reply:
I see. Thanks! 👍